aad/azure-spring-boot-sample-active-directory-b2c-oidc/README.md

```diff
@@ -0,0 +1,112 @@
+5. Replace `${your-logout-success-url}` to `http://localhost:8080/login`.
+
+```yaml
+azure:
+  activedirectory:
+    b2c:
+      base-uri: ${your-tenant-authorization-server-base-uri}
+      client-id: ${your-client-id}
+      client-secret: ${your-client-secret}
+      login-flow: ${your-login-user-flow-key}               # default to sign-up-or-sign-in, will look up the user-flows map with provided key.
+      logout-success-url: ${your-logout-success-url}
+      user-flows:
+        password-reset: ${your-profile-edit-user-flow}
+        profile-edit: ${your-password-reset-user-flow}
+        sign-up-or-sign-in: ${your-sign-up-or-in-user-flow}
+      user-name-attribute-name: ${your-user-name-claim}
+```
+
+**NOTE**: If both `tenant` and `baseUri` are configured at the same time, only `baseUri` takes effect.
+
+### Run with Maven
+```
+cd azure-spring-boot-samples/aad/azure-spring-boot-sample-active-directory-b2c-oidc
+mvn spring-boot:run
+```
+
+### Validation
+
+1. Access `http://localhost:8080/` as index page.
+2. Sign up/in.
+3. Profile edit.
+4. Password reset.
+5. Log out.
+6. Sign in.
+
+## Troubleshooting
+- `Missing attribute 'name' in attributes `
+
+  ```
+  java.lang.IllegalArgumentException: Missing attribute 'name' in attributes
+  	at org.springframework.security.oauth2.core.user.DefaultOAuth2User.<init>(DefaultOAuth2User.java:67) ~[spring-security-oauth2-core-5.3.6.RELEASE.jar:5.3.6.RELEASE]
+  	at org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser.<init>(DefaultOidcUser.java:89) ~[spring-security-oauth2-core-5.3.6.RELEASE.jar:5.3.6.RELEASE]
+  	at org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService.loadUser(OidcUserService.java:144) ~[spring-security-oauth2-client-5.3.6.RELEASE.jar:5.3.6.RELEASE]
+  	at org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService.loadUser(OidcUserService.java:63) ~[spring-security-oauth2-client-5.3.6.RELEASE.jar:5.3.6.RELEASE]
+  ```
+
+  While running sample, if error occurs with logs above:
+
+  - make sure that while creating user workflow by following this [guide](https://docs.microsoft.com/azure/active-directory-b2c/tutorial-create-user-flows), for **User attributes and claims** , attributes and claims for **Display Name** should be chosen.
+
+### FAQ
+
+#### Sign in with loops to B2C endpoint ?
+This issue almost due to polluted cookies of `localhost`. Clean up cookies of `localhost` and try it again.
+
+#### More identity providers from AAD B2C login ?
+Follow the guide of [Set up Google account with AAD B2C](https://docs.microsoft.com/azure/active-directory-b2c/active-directory-b2c-setup-goog-app).
+And also available for Amazon, Azure AD, FaceBook, Github, Linkedin and Twitter.
+
+## Next steps
+## Contributing
+<!-- LINKS -->
+[environment_checklist]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/ENVIRONMENT_CHECKLIST.md#ready-to-run-checklist
```

Comment:
Can we move ENVIRONMENT_CHECKLIST.md to this repo?

Reply:
I am afraid not, there are still 42 references in azure-sdk-for-java except for azure-spring-boot-samples